app/assets/javascripts/discourse/app/templates/wizard/step.gjs

```diff
@@ -0,0 +1,58 @@
+import RouteTemplate from "ember-route-template";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import Component from "@glimmer/component";
+import getUrl from "discourse-common/lib/get-url";
+import WizardCanvas from "discourse/static/wizard/components/wizard-canvas";
+import WizardStep from "discourse/static/wizard/components/wizard-step";
+
+export default RouteTemplate(
+  class extends Component {
+    <template>
+      {{#if this.showCanvas}}
+        <WizardCanvas />
+      {{/if}}
+
+      <WizardStep
+        @step={{@model.step}}
+        @wizard={{@model.wizard}}
+        @goNext={{this.goNext}}
+        @goBack={{this.goBack}}
+        @goHome={{this.goHome}}
+      />
+    </template>
+
+    @service router;
+
+    get step() {
+      return this.args.model.step;
+    }
+
+    get showCanvas() {
+      return this.step.id === "ready";
+    }
+
+    @action
+    goNext(response) {
+      const next = this.step.next;
+
+      if (response?.refresh_required) {
+        document.location = getUrl(`/wizard/steps/${next}`);
+      } else if (response?.success && next) {
+        this.router.transitionTo("wizard.step", next);
+      } else if (response?.success) {
+        this.router.transitionTo("discovery.latest");
+      }
```

Comment:
This is the counterpart of the comment above. If `response` is `undefined`, this doesn't end up doing anything